### ISSUE TITLE
When ACH Processor On, Credit Card Fields Hidden on Member Renewal Form

### ISSUE THREAD

**Issue by shaigluskin** [iATSPayments/com.iatspayments.civicrm#73]
Hi!

Site set-up:

CiviCRM 4.5.8
iATS Extension 1.3.1
Drupal 7.34
PHP 5.4.30
MySQL 5.5.42

Site has **two** payment processors enabled:
* iATS Payments Credit Card
* iATS Payments ACH/EFT

*To replicate apparent bug:*
Go to a contact record, click on membership tab and renew an existing membership by clicking on the "more" (on right) and then select "Renew-Credit Card."

The pop-up that comes up is missing the fields for the credit card information.

[Screen Shot](https://www.dropbox.com/s/53a60aafomn8vqr/renewal_missing_cc_fields.jpg?dl=0)

*Work-around*: go to payment processor admin page (civicrm/admin/paymentProcessor?reset=1) and *disable*  iATS Payments ACH/EFT. Now clicking on "Renew-Credit Card" from a membership and the pop-up will include the credit card fields.

Obviously the work-around is not ideal because ACH has to be turned off for the whole site.

Thanks so much for your help in sorting this out.

Shai

**Comment by KarinG** [comment]
I'll have a look at this - today/tomorrow.

**Comment by adixon** [comment]
Yes, I got this also, there's some more patching of core, to do in CRM/Membership/Form/MembershipRenewal.php, similar to the membership (i.e. restrict payment processors to only credit card ones).
